micropython/build.sh

```diff
@@ -4,41 +4,72 @@ set -e
-PREFIX_UPYTH_SRC=${PREFIX_UPYTH_BUILD}/${UPYTH}
 PREFIX_UPYTH_PATCHSET="${PREFIX_UPYTH}/${UPYTH}-patchset"
-PREFIX_UPYTH_MARKERS="$PREFIX_UPYTH_BUILD/markers/"
+
+b_log "Building micropython"
+pushd .
+cd "$PREFIX_UPYTH"
+
 
 #
-# Download and unpack
+# Download
 #
-mkdir -p "$PREFIX_UPYTH_BUILD" "$PREFIX_UPYTH_MARKERS"
-[ -f "$PREFIX_UPYTH/${UPYTH}.tar.xz" ] || wget https://github.com/micropython/micropython/releases/download/v${UPYTH_VER}/${UPYTH}.tar.xz -P "$PREFIX_UPYTH"
-[ -d "${PREFIX_UPYTH_SRC}" ] || tar xf "$PREFIX_UPYTH/${UPYTH}.tar.xz" -C "$PREFIX_UPYTH_BUILD"
+[ -f "${UPYTH}.tar.xz" ] || wget https://github.com/micropython/micropython/releases/download/v${UPYTH_VER}/${UPYTH}.tar.xz
+[ -d "${UPYTH}" ] || ( echo "Unpacking ${UPYTH}.tar.xz..." && tar xf "${UPYTH}.tar.xz" )
 
 
 #
-# Apply patches
+# Patch
+#
+if ! patch -R -p0 -s -f --dry-run < "${PREFIX_UPYTH_PATCHSET}"/lib.patch >/dev/null 2>&1
+then
+	patch -s -p0 < "${PREFIX_UPYTH_PATCHSET}"/lib.patch && echo "micropython build: lib/ patched"
+	patch -s -p0 < "${PREFIX_UPYTH_PATCHSET}"/py.patch&& echo "micropython build: py/ patched"
+	patch -s -p0 < "${PREFIX_UPYTH_PATCHSET}"/mpy-cross.patch && echo "micropython build: mpy-cross/ patched"
+	patch -s -p0 < "${PREFIX_UPYTH_PATCHSET}"/os.patch && echo "micropython build: ports/unix/ patched"
+fi
+
+
 #
-# FIXME: prefix patch files with numbers to ensure correct order
-for patchfile in "$PREFIX_UPYTH_PATCHSET"/*.patch; do
-	if [ ! -f "$PREFIX_UPYTH_MARKERS/$(basename "$patchfile").applied" ]; then
-		echo "applying patch: $patchfile"
-		patch -d "$PREFIX_UPYTH_SRC" -p1 < "$patchfile"   # FIXME: should be -p1
-		touch "$PREFIX_UPYTH_MARKERS/$(basename "$patchfile").applied"
-	fi
-done
+# Architecture specific flags/values
+#
+export PHOENIX_MATH_ABSENT="expm1 log1p asinh acosh atanh erf tgamma lgamma copysign __sin __cos __tan __signbit"
+if [[ ${UPYTH_CC_BASE} == "arm-phoenix-" ]]
+then
+	UPYTH_STACKSZ="2048"
+	UPYTH_HEAPSZ="16384"
+	export LDFLAGS_EXTRA=" -Wl,-q "
+	export CFLAGS_EXTRA="-DUPYTH_STACKSZ=${UPYTH_STACKSZ} -DUPYTH_HEAPSZ=${UPYTH_HEAPSZ} -nostartfiles -fpic -fpie -msingle-pic-base -mno-pic-data-is-text-relative "
+	export STRIPFLAGS_EXTRA="--strip-unneeded"
+	export PHOENIX_PORT_FLAGS=${CFLAGS}
+elif [[ ${UPYTH_CC_BASE} == "i386-pc-phoenix-" ]]
+then
+	UPYTH_STACKSZ="65536"
+	UPYTH_HEAPSZ="65536"
+	export CFLAGS_EXTRA=" -DUPYTH_STACKSZ=${UPYTH_STACKSZ} -DUPYTH_HEAPSZ=${UPYTH_HEAPSZ} "
+	export STRIPFLAGS_EXTRA="--strip-unneeded"
+else
+	echo "Error: there is no micropython port for chosen architecture"
+	exit 1
+fi
+
+
+#
+# Clean and build
+#
+([ -n "$CLEAN" ] && (cd "${UPYTH}/mpy-cross" && make clean) && (cd "${UPYTH}/ports/unix" && make clean))
+(cd "${UPYTH}/mpy-cross" && make all BUILD="${PREFIX_UPYTH_BUILD}" CROSS_COMPILE="${UPYTH_CC_BASE}")
+(cd "${UPYTH}/ports/unix" && make all CROSS_COMPILE="${UPYTH_CC_BASE}" && "${UPYTH_CC_BASE}"strip ${STRIPFLAGS_EXTRA} micropython)
```

Comment:
install not stripped binary to `PREFIX_PROG` and then strip it into `PREFIX_PROG_STRIPPED` (unstripped binary is needed for debugging or at least `addr2line` to work correctly)

Reply:
Reverted to previous form (where those changes were already in place)